### ISSUE TITLE
[Bug Report][2.5.0] can't initiate state after calling form validate function

### ISSUE THREAD

**Issue by Masber** [vuetifyjs/vuetify#13705]
### Environment
**Vuetify Version:** 2.5.0
**Vue Version:** 2.6.11
**Browsers:** Chrome 90.0.4430.212
**OS:** Mac OS 10.15.7

### Steps to reproduce
click "RESET FORM" button

### Expected Behavior
application state should be initialized after resetting the form

### Actual Behavior
both name and email fields are empty after resetting the form

### Reproduction Link
<a href="https://codepen.io/masber/pen/OJpgxBo?editors=101" target="_blank">https://codepen.io/masber/pen/OJpgxBo?editors=101</a>

### Other comments
I copied this example from vuetify docs website and modified for this purpose. The important thing is that the reset method calls another method "initState" after resetting the form. This functionality may be useful for generic components that will be reused many times with different init values that may reshape the component.

**Comment by K-Schaeffer** [comment]
I couldn't reproduce the issue, in my tests It worked as expected: both name and email fields having their states initialized again.

One thing I did to ensure that is working was removing the the initState call on mounted hook and assigning a value both to email and name in the data property. Then, I clicked on the reset button, and again, it worked.

Let me know if I misunderstood something or just tested it wrongly.
